crates/shared/src/price_estimation/instrumented.rs

```diff
@@ -37,15 +37,18 @@ impl<T> InstrumentedPriceEstimator<T> {
         }
     }
 
-    /// Determines the result of a price estimate as either "success" or
-    /// "failure".
+    /// Determines the result of a price estimate, returning either "success" or
+    /// the error reason
     fn estimate_result<B>(&self, estimate: Result<&B, &PriceEstimationError>) -> &str {
         // Count as a successful request if the answer is ok (no error) or if the error
         // is No Liquidity
-        if estimate.is_ok() || matches!(estimate, Err(PriceEstimationError::NoLiquidity)) {
-            "success"
-        } else {
-            "failure"
+        match estimate {
+            Ok(_) | Err(PriceEstimationError::NoLiquidity) => "success",
```

Comment:
Given that the intention of this PR is to expose a helpful dashboard for solver about their quoting performance I think it makes more sense to have a separate label for `NoLiquidity` and adjust grafana queries that currently use `success` to use `success|no_liqudity` going forward.